bin/problem.py

```diff
@@ -223,34 +160,22 @@ def __init__(
         self.generator_time: int = parse_setting(yaml_data, "generator_time", 60)  # in seconds
         self.visualizer_time: int = parse_setting(yaml_data, "visualizer_time", 60)  # in seconds
 
-        # Try to read deprecated ways of setting the time limit.
-        def _get_legacy_time_limit():
-            timelimit_path = problem.path / ".timelimit"
-            if timelimit_path.is_file():
-                if not problem_settings.is_legacy():
-                    log("A .timelimit file is DEPRECATED. Use limits.time_limit instead.")
-                return float(timelimit_path.read_text())
-
-            domjudge_path = problem.path / "domjudge-problem.ini"
-            if domjudge_path.is_file():
-                log("domjudge-problem.ini is DEPRECATED. Use limits.time_limit instead.")
-                for line in domjudge_path.read_text().splitlines():
-                    key, var = map(str.strip, line.strip().split("="))
-                    if (var[0] == '"' or var[0] == "'") and (var[-1] == '"' or var[-1] == "'"):
-                        var = var[1:-1]
-                    if key == "timelimit":
-                        return float(var)
-
-        # If limits.time_limit does not exist, attempt to use legacy_time_limit instead.
-        legacy_time_limit = _get_legacy_time_limit()
-        self.time_limit: float = time_limit or legacy_time_limit or 1.0
-        self.time_limit_is_default: bool = time_limit is None and legacy_time_limit is None
+        # warn for deprecated timelimit files
+        if (problem.path / ".timelimit").is_file():
+            warn("A .timelimit file is DEPRECATED. Use limits.time_limit instead.")
+        if (problem.path / "domjudge-problem.ini").is_file():
+            warn(
+                "domjudge-problem.ini is DEPRECATED. Use limits.time_limit if you want to set a timelimit."
+            )
```

Comment:
If we're going for a hard cut, I think we can remove this as well :stuck_out_tongue:

Reply:
i think a warning os still nice for now